### ISSUE TITLE
Provide CancellationToken as optional parameter to all methods, so requests can be cancelled or timed-out.

### ISSUE THREAD

**Issue by pianomanjh** [nozzlegear/ShopifySharp#238]
It is common for APIs involving potentially long http requests, to accept a `CancellationToken`, such that a call can be cancelled, or timed-out by a user.
It would be fairly simple to add to this API, as it would be just need be passed down to the `HttpClient`, that accepts one in `SendAsync`.

**Comment by nozzlegear** [comment]
Good idea! In your mind, how would this look? An overload for each function on a service that accepts an optional token?

```cs
public List<Something> ListAsync(ListOptions options, CancellationToken token = null)
{
...
}
```

**Comment by pianomanjh** [comment]
yes, any method that results in an HttpRequest could accept an optional token.

**Comment by josh-degraw** [comment]
Any traction on this? This could be very helpful in my current use-cases.

IMO It would also be useful to update the `ExequteRequestAsync<T>` delegate  and IRequestExecutionPolicy to either pass in a cancellation token as is, or add an overload that would have a CancellationToken, e.g. the Retry-able execution policies could support cancellation to avoid performance issues at runtime.

**Comment by nozzlegear** [comment]
I’m still open to this! If anyone has the time I’d be happy to review a pull request to get this added in.

**Comment by pianomanjh** [comment]
Alright, I can take a stab at this, standby

**Comment by pianomanjh** [comment]
PR posted: https://github.com/nozzlegear/ShopifySharp/pull/476

**Comment by nozzlegear** [comment]
Published in 5.1.0 :rocket:
